scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
 
     def __eq__(self, other):
         return self.url == other.url and self.text == other.text and \
```

Comment:
Hm, why are they detected as cp1252? From reading https://github.com/scrapy/scrapy/blob/afac3fd2c28ac5b16e7e3c95fca5acd1334f8c58/scrapy/http/response/text.py#L96 I got an idea it'll try ascii first, then try utf8, and only then cp1252, if ascii and utf8 fails